core/appeal/service.go

```diff
@@ -1098,44 +1127,31 @@ func (s *Service) CreateAccess(ctx context.Context, a *domain.Appeal, opts ...Cr
 		}
 	}
 
-	if err := s.providerService.GrantAccess(ctx, a); err != nil {
+	if err := s.providerService.GrantAccess(ctx, *a.Access); err != nil {
 		return fmt.Errorf("granting access: %w", err)
 	}
 
-	if err := a.Activate(); err != nil {
-		return fmt.Errorf("activating appeal: %w", err)
-	}
-
 	return nil
 }
 
-func (s *Service) isEligibleToExtend(a *domain.Appeal, p *domain.Provider, activeAppealsMap map[string]map[string]map[string]*domain.Appeal) (bool, error) {
-	if activeAppealsMap[a.AccountID] != nil &&
-		activeAppealsMap[a.AccountID][a.ResourceID] != nil &&
-		activeAppealsMap[a.AccountID][a.ResourceID][a.Role] != nil {
+func (s *Service) isEligibleToExtend(a *domain.Appeal, p *domain.Provider, activeAccesses map[string]map[string]map[string]*domain.Access) (bool, error) {
```

Comment:
will rename this to `checkExtensionEligibility`

Reply:
> Is there any benefit to return bool here?

yes is a possibility where it returns no error (all dependencies information are complete/no missing values) but still not eligible to extend because of the policy

Reply:
Why not returning something like `ErrNotEligible` ?